### ISSUE TITLE
When to call mixpanel.init() / when to call the constructor new Mixpanel('--token--');

### ISSUE THREAD

**Issue by Spark-Cybel** [mixpanel/mixpanel-react-native#104]
We are integrating mixpanel into our legacy app.
I started POC.
And I added this on the home screen home.js
But the problem is whenever I land up on the home screen
The new distinct id is getting assigned to all events.

**How to avoid that without setting up distinct id explicitly?**

or
### **What is the correct place and correct way to call the constructor and init methods?**

**home.js** File
```

..........
..........

import { Mixpanel } from 'mixpanel-react-native';

..........

const mixpanel = new Mixpanel('--token--');
mixpanel.init();
mixpanel.getDistinctId()
  .then((id) => console.log('===>>> DistinctId ', id));

const Home = (props) => {

..........
..........

  return(
      <View>
        ..........
            <Button
              title='Select Premium Plan'
              onPress={ () => mixpanel.track('Test1', {'status': 'working'}) }
            />
        ..........
        ..........
        ..........
      </View>
  );
};

```

### Dos and Don'ts inputs for mixpanel integration will be really helpful!! 🙌🙂🤞

You can see the two different distinct ids were assigned
Step I did:-
1. Clicked on a button "Test 1" event gets logged.
2. Relaunched the App and again clicked on a button "Test1" event gets logged
But with the different distinct ID

![Screenshot 2021-10-07 at 1 26 53 PM](https://user-images.githubusercontent.com/89348687/136343479-f78d9126-aad2-40b9-a51d-c0d207914411.png)

**Comment by zihejia** [comment]
hi @Spark-Cybel , we will release a new version addressing this issue in early Jan after the holiday. Sorry about the inconvenience.

**Comment by zihejia** [comment]
hi @Spark-Cybel ,  again, thanks for bringing it up. This issue has been fixed in [1.3.5](https://github.com/mixpanel/mixpanel-react-native/releases/tag/v1.3.5). The distinct id is not going to be changed for new app sessions. I'm closing this issue now, please feel free to reopen if the problem still exists.
